samples/snapshot/snapshot_sample.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+from opensearchpy import OpenSearch
+
+# connect to OpenSearch
+
+host = 'localhost'
+port = 9200
+auth = ('admin', 'admin') # For testing only. Don't store credentials in code.
+
+client = OpenSearch(
+    hosts = [{'host': host, 'port': port}],
+    http_auth = auth,
+    use_ssl = True,
+    verify_certs = False,
+    ssl_show_warn = False
+)
+
+# Create an index 
+
+index_name = "test-snapshot"
+client.indices.create(index = index_name)
+
+# Create a snapshot repository
+
+repo_body = {
+    "type": "fs",  # Replace 'fs' with the appropriate repository type
+    "settings": {
+        "location": "/path/to/repo",  # Replace with the desired repository location
```

Comment:
I am not really sure if I need to keep this as a location example for it to be replaced by the client or if is there a way to retrieve the default location for a file system repository.

Reply:
Would `tempfile.TemporaryDirectory` work for the sample?

Reply:
For windows may be use `cd opensearch-py/samples/`

Reply:
I apologize for the wrong screenshot. Yes, I have tried it within samples directory. I get the following error: 
$ poetry run hello/hello.py
"hello" is not an internal or external
command, executable program, or batch file.
(base)

And I am running this on my main branch, which is up-to-date with my remote origin. Could you please let me know if there are any pre-steps I need to take besides the ones on ./samples/README.md?

Reply:
Hello @roma2023,

I successfully followed the steps on my Mac without encountering any errors:

1. Installed Poetry in the home directory using `pipx install poetry`.
2. Navigated to the "opensearch-py/samples" directory. `cd opensearch-py/samples`
3. Installed the project dependencies with `poetry install`.
4. Executed `poetry run hello/hello.py`.
5. Please uninstall Poetry and reattempt the steps mentioned above. If you encounter any further issues, please don't hesitate to let me know.

Reply:
Hi @saimedhi, 

I went through the steps over again, but still receiving the same error. I will try to make an issue on this, I believe it is due to the system configuration. 

Meanwhile, could you please try testing both:

1. ```poetry run snapshot\snapshot_sample``` on this version of the PR code 
2.   And the same command, but with the ```tempfile.TemporaryDirectory``` instead of ```path\to\repo``` for the location as mentioned above. 

Please let us know the results. Thank you)

Reply:
please make changes related to tempfile.TemporaryDirectory and also add imports as required. I am happy to verify your PR but before than please raise an issue with screenshots that you are facing it helps other contributors as well.
Thanks.

Reply:
@roma2023 Thanks a lot for your patience.

Reply:
@saimedhi Done! I opened an issue and updated the sample per your request.

Reply:
`poetry run python snapshot/snapshot_sample.py`
```
Traceback (most recent call last):
  File "/Users/saimedhi/code_pythondiff/opensearch-py/samples/snapshot/snapshot_sample.py", line 43, in <module>
    response = client.snapshot.create_repository(repository = repository_name, body = repo_body)
  File "/Users/saimedhi/Library/Caches/pypoetry/virtualenvs/package-khtnLQLT-py3.10/lib/python3.10/site-packages/opensearchpy/client/utils.py", line 179, in _wrapped
    return func(*args, params=params, headers=headers, **kwargs)
  File "/Users/saimedhi/Library/Caches/pypoetry/virtualenvs/package-khtnLQLT-py3.10/lib/python3.10/site-packages/opensearchpy/client/snapshot.py", line 185, in create_repository
    return self.transport.perform_request(
  File "/Users/saimedhi/Library/Caches/pypoetry/virtualenvs/package-khtnLQLT-py3.10/lib/python3.10/site-packages/opensearchpy/transport.py", line 409, in perform_request
    raise e
  File "/Users/saimedhi/Library/Caches/pypoetry/virtualenvs/package-khtnLQLT-py3.10/lib/python3.10/site-packages/opensearchpy/transport.py", line 370, in perform_request
    status, headers_response, data = connection.perform_request(
  File "/Users/saimedhi/Library/Caches/pypoetry/virtualenvs/package-khtnLQLT-py3.10/lib/python3.10/site-packages/opensearchpy/connection/http_urllib3.py", line 266, in perform_request
    self._raise_error(
  File "/Users/saimedhi/Library/Caches/pypoetry/virtualenvs/package-khtnLQLT-py3.10/lib/python3.10/site-packages/opensearchpy/connection/base.py", line 301, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(
opensearchpy.exceptions.TransportError: TransportError(500, 'repository_exception', "[my_repository] location [/var/folders/fk/8ht5889j6nb3zqzygn4qcfqc0000gr/T/tmparzg0g99] doesn't match any of the locations specified by path.repo because this setting is empty")
```

Reply:
@saimedhi So, I see ```tempfile.TemporaryDirectory``` is failing to make an existing directory. Can you please try running the previous version (1 commit behind) of the code with the existing location on your machine instead of ```path/to/repo``` for location?

Reply:
@saimedhi I have attempted to run the sample with different existing directory locations on my machine, but I keep having the following trace: 
``` $ poetry run python snapshot/snapshot_sample.py
Traceback (most recent call last):
  File "C:\Users\HP\Desktop\AmazonOpenSearchService\Opensearch-py-ml\opensearch-py\samples\snapshot\snapshot_sample.py", line 39, in <module>
    response = client.snapshot.create_repository(repository = repository_name, body = repo_body)
  File "C:\Users\HP\AppData\Local\pypoetry\Cache\virtualenvs\package-yvpjSqWs-py3.9\lib\site-packages\opensearchpy\client\utils.py", line 179, in _wrapped
    return func(*args, params=params, headers=headers, **kwargs)
  File "C:\Users\HP\AppData\Local\pypoetry\Cache\virtualenvs\package-yvpjSqWs-py3.9\lib\site-packages\opensearchpy\client\snapshot.py", line 185, in create_repository
    return self.transport.perform_request(
  File "C:\Users\HP\AppData\Local\pypoetry\Cache\virtualenvs\package-yvpjSqWs-py3.9\lib\site-packages\opensearchpy\transport.py", line 409, in perform_request
    raise e
  File "C:\Users\HP\AppData\Local\pypoetry\Cache\virtualenvs\package-yvpjSqWs-py3.9\lib\site-packages\opensearchpy\transport.py", line 370, in perform_request
    status, headers_response, data = connection.perform_request(
  File "C:\Users\HP\AppData\Local\pypoetry\Cache\virtualenvs\package-yvpjSqWs-py3.9\lib\site-packages\opensearchpy\connection\http_urllib3.py", line 266, in perform_request       
    self._raise_error(
  File "C:\Users\HP\AppData\Local\pypoetry\Cache\virtualenvs\package-yvpjSqWs-py3.9\lib\site-packages\opensearchpy\connection\base.py", line 301, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(
opensearchpy.exceptions.TransportError: TransportError(500, 'repository_exception', "[my_repository] location [./] doesn't match any of the locations specified by path.repo because this setting is empty")
(base)
 ```

It keeps stating: ```location [./] doesn't match any of the locations specified by path.repo because this setting is empty ``` 

I could not find any ```path.repo``` file in the project. 

I believe there might be a need for building a ```path.repo``` setting with some default system locations and use them in samples. What do you think about this issue?

Reply:
@dblock Could you please take a look at this issue?

Reply:
relevant doc. https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore/#shared-file-system opensearch.yml file will be found in docker.

Reply:
@dblock How can I run the docker with the OpenSearch server write permission?

Reply:
I am trying to run ```snapshot``` volume following the steps specified here: https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore/, 

but the ```poetry run python snapshot/snapshot_sample``` fails with the following error: 
```
$ poetry run python snapshot/snapshot_sample.py
Traceback (most recent call last):
  File "C:\Users\HP\Desktop\AmazonOpenSearchService\Opensearch-py-ml\opensearch-py\samples\snapshot\snapshot_sample.py", line 39, in <module>
    response = client.snapshot.create_repository(repository = repository_name, body = repo_body)
  File "C:\Users\HP\AppData\Local\pypoetry\Cache\virtualenvs\package-yvpjSqWs-py3.9\lib\site-packages\opensearchpy\client\utils.py", line 179, in _wrapped
    return func(*args, params=params, headers=headers, **kwargs)
  File "C:\Users\HP\AppData\Local\pypoetry\Cache\virtualenvs\package-yvpjSqWs-py3.9\lib\site-packages\opensearchpy\client\snapshot.py", line 185, in create_repository
    return self.transport.perform_request(
  File "C:\Users\HP\AppData\Local\pypoetry\Cache\virtualenvs\package-yvpjSqWs-py3.9\lib\site-packages\opensearchpy\transport.py", line 409, in perform_request
    raise e
  File "C:\Users\HP\AppData\Local\pypoetry\Cache\virtualenvs\package-yvpjSqWs-py3.9\lib\site-packages\opensearchpy\transport.py", line 370, in perform_request
    status, headers_response, data = connection.perform_request(
  File "C:\Users\HP\AppData\Local\pypoetry\Cache\virtualenvs\package-yvpjSqWs-py3.9\lib\site-packages\opensearchpy\connection\http_urllib3.py", line 266, in perform_request
    self._raise_error(
  File "C:\Users\HP\AppData\Local\pypoetry\Cache\virtualenvs\package-yvpjSqWs-py3.9\lib\site-packages\opensearchpy\connection\base.py", line 301, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(
opensearchpy.exceptions.TransportError: TransportError(500, 'repository_verification_exception', 'failed to create blob container')
```
I believe this is because docker is denying the write permission.

Reply:
I don't know the details of how this feature works, but I imagine that the docker container will need something like `--env path.repo=/tmp`. It would be totally fine to document that to run this sample you need to create a docker container in a certain way.